src/ethereum/arrow_glacier/vm/instructions/system.py

```diff
@@ -71,6 +71,10 @@ def generic_create(
     # if it's not moved inside this method
     from ...vm.interpreter import STACK_DEPTH_LIMIT, process_create_message
 
+    call_data = memory_read_bytes(
+        evm.memory, memory_start_position, memory_size
+    )
+
```

Comment:
Why move this up here?

Reply:
When `EIP-3860` was implemented in Shanghai, we had to move the `cal_data` calculation up (before adding the contract address to the `accessed_address`) to check for cases where the `initcode` was too long. We had just missed back-porting this change to the older forks at the time. This fixes that miss.